### PR TITLE
doc(task-library): document discover requirement for join-up

### PR DESCRIPTION
**Pull request by zehicle**
clarify something from community discussion

### CODE REVIEW

File: task-library/workflows/discover-joinup.yaml [hunk 1]
```diff
@@ -5,6 +5,11 @@ Documentation: |
 
   This workflow is recommended for joining cloud machines instead of `discover-base`.
 
+  NOTE: You must set `DefaulBootEnv` to `discover` in order to use join-up
```
Comment (galthaus) [inline task-library/workflows/discover-joinup.yaml:8]: defaultBootEnv 

the value should be sledgehammer. 

discovery is used for the unknownBootEnv.  This is the more likely solution for general use for creating machines.

Comment (zehicle) [reply task-library/workflows/discover-joinup.yaml]: updated

File: task-library/workflows/discover-joinup.yaml [hunk 2]
```diff
@@ -5,6 +5,11 @@ Documentation: |
 
   This workflow is recommended for joining cloud machines instead of `discover-base`.
 
+  NOTE: You must set `defaultBootEnv` to `sledgehammer` in order to use join-up
+  to discover machines.  Alternatively, you can create placeholder
+  machines with `.BootEnv: discover` and use `join-up.sh` to connect to the placeholder
```
Comment (galthaus) [inline task-library/workflows/discover-joinup.yaml:10]: discover needs to be sledgehammer here.  We don't have discover BootEnv.

Also, I don't know this works.  I think discovery (no machine) works only if unknownBootEnv is set to discovery.

I don't know how or where join-up.sh works to know for sure.

Comment (zehicle) [reply task-library/workflows/discover-joinup.yaml]: ok, tried to update.

**Pull request merged**